common/src/main/java/org/broadleafcommerce/common/config/BroadleafCommonConfig.java

```diff
@@ -48,4 +65,64 @@ public JpaVendorAdapter blJpaVendorAdapter() {
         return vendorAdapter;
     }
 
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public Compiler blClosureCompiler() {
+        return new Compiler();
+    }
+
+    @Bean
+    public CompilerOptions blClosureCompilerOptions() {
+        CompilerOptions options = new CompilerOptions();
+
+        options.setCodingConvention(new ClosureCodingConvention());
+
+        if (StringUtils.isNoneBlank(compilerLanguageIn)) {
+            CompilerOptions.LanguageMode languageIn = CompilerOptions.LanguageMode.valueOf(compilerLanguageIn);
+            if (StringUtils.isBlank(compilerLanguageOut)) {
+                options.setLanguage(languageIn);
+            } else {
+                options.setLanguageIn(languageIn);
+            }
+        }
+
+        if(StringUtils.isNoneBlank(compilerLanguageOut)) {
+            options.setLanguageOut(CompilerOptions.LanguageMode.valueOf(compilerLanguageOut));
+        }
+
+        CompilationLevel.SIMPLE_OPTIMIZATIONS.setOptionsForCompilationLevel(options);
+
+        WarningLevel.valueOf(warningLevel).setOptionsForWarningLevel(options);
+
+        options.setClosurePass(true);
+        options.setRewritePolyfills(toFeatureSet(options.getLanguageIn()).contains(FeatureSet.ES6));
+        options.setStrictModeInput(true);
+
+        return options;
+    }
+
+    protected FeatureSet toFeatureSet(CompilerOptions.LanguageMode languageMode) {
+        switch (languageMode) {
+            case ECMASCRIPT3:
+                return FeatureSet.ES3;
+            case ECMASCRIPT5:
+            case ECMASCRIPT5_STRICT:
+                return FeatureSet.ES5;
+            case ECMASCRIPT_2015:
+                return FeatureSet.ES6_MODULES;
+            case ECMASCRIPT_2016:
+                return FeatureSet.ES7_MODULES;
+            case ECMASCRIPT_2017:
+                return FeatureSet.ES8_MODULES;
+            case ECMASCRIPT_NEXT:
+                return FeatureSet.ES_NEXT;
+            case ECMASCRIPT6_TYPED:
+                return FeatureSet.TYPESCRIPT;
+            case NO_TRANSPILE:
+                throw new IllegalStateException();
+        }
+
+        throw new IllegalStateException();
+    }
+
```

Comment:
All of this should be in an `@Conditional` on a closure-compiler class existing since we're making it an optional dependency.

Reply:
Sorry if the code snippet is vague. I'm referring to all of the Google Closure configuration